src/classes/providers/shifter/types.ts

```diff
@@ -0,0 +1,33 @@
+import { AxiosRequestConfig } from 'axios';
+import { IProviderConfig, EStrategyMode } from '../base/types';
+
+type TShifterStrategy = TShifterStrategyChangeIpEveryRequest;
+
+enum EShifterCountry {
+  FRANCE = 'fr',
+  BELGIUM = 'be',
+  SWITZERLAND = 'ch',
+}
+
+interface IShifterCountryPortMapping {
+  fr?: number[];
+  be?: number[];
+  ch?: number[];
+}
```

Comment:
Apparently typescript is not "smart" enough to general this interface based on the enum up above. But maybe there's a clever way to deal with this? I'm up for suggestions if it seems necessary.

Reply:
if you want to have a typed key you need to use a `Record` i think

Reply:
`Partial<Record<enum, type>>` did the trick 👍